src/components/EventsTimeline/interfaces.ts

```diff
@@ -0,0 +1,59 @@
+import { CogniteEvent } from '@cognite/sdk';
+import { SyntheticEvent } from 'react';
+
+export enum EventTimelineType {
+  discrete = 'discrete',
+  continuous = 'continuous',
+}
+
+export enum EventTimelineView {
+  fill = 'fill',
+  outline = 'outline',
+}
+
+export interface TimelineEventAppearance {
+  view: EventTimelineView;
+  type: EventTimelineType;
+}
+
+export interface TimelineEvent extends TimelineEventAppearance {
+  id: number;
+}
+
+export interface CogniteEventForTimeline extends CogniteEvent {
+  appearance: TimelineEventAppearance;
+  color?: string;
+}
+
+export interface TimelineRuler {
+  show: boolean;
+  onChange?: (
+    e: SyntheticEvent,
+    date: number,
+    events?: CogniteEventForTimeline[]
+  ) => void;
```

Comment:
not sure here, but `onRulerChange(...) takes 3 parameters, and I am thinking maybe to wrap it in object, to make it less messy? Then we can define a type for that

also the `date` paramater is actually a timestamp right? probably rename it?

Reply:
Yep, this is a good point, I'll add an interface for this